.github/workflows/ci.yml

```diff
@@ -0,0 +1,49 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Continuous integration
+
+# Run the CI on any direct push to a branch on this repository or on pull request
+# branches
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10"]
```

Comment:
That's a lot of pythons - how long does it take?

Reply:
Well those are the Pythons we support. Usually, you will run CI on all the Python version you support, to make sure that you didn't inadvertently used syntax which is exclusive to newer Pythons or that one of our dependencies have stopped supporting an older version. To answer your question, minutes, usually, a few  and rarely no more than you can count on fingers. It is a little difficult to answer precisely, because using the free service, we are of course on shared HW which by design has a high load, so sometimes you have to "wait your turn". However, I have never observed this practically to be a problem, because the cases where you really want CI results visible on GH, is when you are handing a PR over for review, which almost always involves another person, which again usually have a hand off latency of more than a few minutes ;)

The hidden agenda here is, that with the ability to run CI (especially tests) on all supported Pythons automatically on GH's systems, we essentially remove the need for tox, which mostly served that purpose, just locally, where actually getting all those Pythons installed is a hassle anyway. So if this is merged, I will shortly propose to remove tox, to simplify tooling somewhat.